### PR TITLE
applications: nrf_desktop: Document new configuration options

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces documentation for new configuration options and release notes.

Jira: NCSDK-17315

### CODE REVIEW

**Review by kapi-no — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -942,6 +942,59 @@ You must modify these configuration sources when `Adding a new board`_, as descr
...
+
+Bluetooth configuration
+-----------------------
+
+The nRF Desktop application introduces application-specific configuration options related to Bluetooth connectivity configuration.
+These options are defined in :file:`Kconfig.ble` file.
+
+The :ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>` Kconfig option enables support for Bluetooth connectivity in the nRF Desktop application.
+The option is enabled by default.
+The option:
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:983]:
```suggestion
The option performs the following:
```

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -942,6 +942,59 @@ You must modify these configuration sources when `Adding a new board`_, as descr
...
+
+* enables application modules related to Bluetooth that are required for the selected device role,
+* enables required functionalities in Zephyr's Bluetooth stack,
+* overlays Bluetooth Kconfig option defaults to align them with the nRF Desktop use-case.
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:987]:
```suggestion
* Enables application modules related to Bluetooth that are required for the selected device role
* Enables required functionalities in Zephyr's Bluetooth stack
* Overlays Bluetooth Kconfig option defaults to align them with the nRF Desktop use-case
```

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -942,6 +942,59 @@ You must modify these configuration sources when `Adding a new board`_, as descr
...
+
+See :file:`Kconfig.ble` file content for details.
+
+CAF configuration
+-----------------
+
+The nRF Desktop application overlays defaults of :ref:`lib_caf` related Kconfig options to align them with the nRF Desktop use-case.
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:994]:
```suggestion
The nRF Desktop application overlays defaults of the :ref:`lib_caf` related Kconfig options to align them with the nRF Desktop use-case.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -207,6 +207,10 @@ nRF Desktop
 * Introduced :ref:`CONFIG_DESKTOP_USB_REMOTE_WAKEUP <config_desktop_app_options>` Kconfig option for :ref:`nrf_desktop_usb_state`.
   The option enables the USB wakeup functionality in the application.
   The option selects :kconfig:option:`CONFIG_USB_DEVICE_REMOTE_WAKEUP`.
+* The Bluetooth GATT Service Changed (:kconfig:option:`CONFIG_BT_GATT_SERVICE_CHANGED`) is disabled on nRF Desktop dongles to reduce memory footprint.
+* Added application-specific Kconfig options to simplify configuration.
```
Comment (annwoj) [inline doc/nrf/releases/release-notes-changelog.rst:211]:
```suggestion
* Added application-specific Kconfig options to simplify the configuration.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -207,6 +207,10 @@ nRF Desktop
...
+  Part of application Kconfig configuration that is common for the selected HID device type is introduced as overlays for default Kconfig values.
```
Comment (annwoj) [inline doc/nrf/releases/release-notes-changelog.rst:212]:
```suggestion
  Part of an application Kconfig configuration that is common for the selected HID device type is introduced as overlays for default Kconfig values.
```

File: applications/nrf_desktop/README.rst [hunk 6]
```diff
@@ -942,6 +942,59 @@ You must modify these configuration sources when `Adding a new board`_, as descr
...
+For example, :ref:`nrf_desktop_hid_state` is automatically enabled for the HID peripheral role.
+Apart from device role, the application-specific Kconfigs specify set of supported HID reports and supported HID boot report (if any).
+
+By default, the nRF Desktop devices use predefined format of HID reports.
+The common HID report map is defined in the :file:`configuration/common/hid_report_desc.c` file.
+
+Default common configuration
+----------------------------
+
+The nRF Desktop application overlays Kconfig defaults and selects Kconfig options required in all of the configurations to align the configuration with the nRF Desktop use-case.
```
[the remark this answers is not in the document]
Comment (annwoj) [reply applications/nrf_desktop/README.rst:971]: @MarekPieta I changed this sentence a bit, could you please double-check its meaning?

**Review by annwoj — CHANGES_REQUESTED** [review]
Added few suggestions

**Review by annwoj — APPROVED** [review]

**Pull request merged**